bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/link/LinkBuilderImpl.java

```diff
@@ -156,7 +162,12 @@ public LinkBuilderImpl(String url, SlingHttpServletRequest req, List<PathProcess
     private @NotNull Link buildLink(String path, SlingHttpServletRequest request, Map<String, String> htmlAttributes) {
         if (StringUtils.isNotEmpty(path)) {
             try {
-                path = URLDecoder.decode(path, StandardCharsets.UTF_8.name());
+                // The link contain character sequences that are not well formatted and cannot be decoded, for example
+                // Adobe Campaign expressions like: /content/path/to/page.html?recipient=<%= recipient.id %>
+                Map<String, String> placeholders = new LinkedHashMap<>();
+                String maskedPath = mask(path, placeholders);
+                maskedPath = URLDecoder.decode(maskedPath, StandardCharsets.UTF_8.name());
+                path = unmask(maskedPath, placeholders);
             } catch (UnsupportedEncodingException e) {
```

Comment:
Maybe we should still catch the IEA and continue processing without decoding. That would also preserve the behaviour in a none-breaking way other, custom cases. Customers that use other markup than the Adobe Campaign expressions would still face the exception.